cfgov/v1/migrations/0226_update_enforcement_action_data.py

```diff
@@ -0,0 +1,42 @@
+# Generated by Django 2.2.13 on 2020-07-13 17:19
+from django.db import migrations
+from v1.models.learn_page import EnforcementActionStatus
+from v1.util.ref import enforcement_statuses
+
+
+def get_value(label):
+    for status in enforcement_statuses:
+        if label == status[1]:
+            return status[0]
+
+
+def forward(apps, schema_editor):
+    for enforcement_status in EnforcementActionStatus.objects.all():
+        page = enforcement_status.action
+        url = page.get_url()
+
+        if not page.live:
+            continue
+        if 'policy-compliance/enforcement/actions' not in url:
+            continue
+        if page.has_unpublished_changes:
+            continue
+
+        existing_status = enforcement_status.status
+        enforcement_status.status = get_value(existing_status)
```

Comment:
This PR updated the `EnforcementActionStatus` model to use different underlying values for the `status` field, while the display names stayed the same (new values are listed in `v1/util/ref.py`). Now, an object that had the status value `Post Order/Post Judgment` in the database should have `post-order-post-judgment` instead.

Seems to me that a data migration is the simplest and most reliable way to make that update, but I'd like a second opinion. @chosak or anyone else, does this seem reasonable to you?

Reply:
A data migration seems reasonable to me.

One potential performance optimization would be to do something like `EnforcementActionStatus.objects.prefetch_related('action')` -- this would retrieve all status pages from the DB in a single query, instead of doing one page query per status in the loop.

Thinking further, is there any reason why you don't want to bulk update _all_ EASes, including any related to unpublished pages or pages elsewhere in the tree? That could maybe be accomplished with a simpler set of commands using [`bulk_update`](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#bulk-update), something like (untested):

```py
for before, after in (
    ('Expired/Terminated/Dismissed', 'expired-terminated-dismissed'),
    ('Pending Litigation', 'pending-litigation'),
    ('Post Order/Post Judgment', 'post-order-post-judgment'),
):
    statuses = EnforcementActionStatus.objects.filter(status=before)
    for status in statuses:
        status.status = after

     # optionally include batch_size here if there are many statuses.
    EnforcementActionStatus.objects.bulk_update(statuses, ['status'])
```

Reply:
Great feedback, thanks. I'll try this out.

Reply:
We talked it through and decided that `bulk_update` is a good option here, so I updated my migration to use it. Thanks for the excellent suggestion.